### PR TITLE
customization #15,22,24

### PR DESCRIPTION
**Pull request by ShashaQin**
simplified time log

### CODE REVIEW

File: erpnext/projects/doctype/time_log/time_log.js [hunk 1]
```diff
@@ -25,16 +25,34 @@ frappe.ui.form.on("Time Log", "refresh", function(frm) {
 
 
 // set to time if hours is updated
-frappe.ui.form.on("Time Log", "hours", function(frm) {
-	if(!frm.doc.from_time) {
-		frm.set_value("from_time", frappe.datetime.now_datetime());
```
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.js]: is this code necessary if "Simplified Time Log" is unchecked?

File: erpnext/projects/doctype/time_log/time_log.json [hunk 2]
```diff
@@ -12,49 +12,51 @@
  "fields": [
   {
    "allow_on_submit": 0, 
-   "bold": 1,
```
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.json]: what did you change on this files? just the fields priority?

File: erpnext/projects/doctype/time_log/time_log.py [hunk 3]
```diff
@@ -58,16 +60,20 @@ def set_status(self):
...
 		if self.production_order:
 			self.title = _("{0} for {1}").format(self.operation, self.production_order)
-		elif self.activity_type and (self.task or self.project):
-			self.title = _("{0} for {1}").format(self.activity_type, self.task or self.project)
-		else:
-			self.title = self.task or self.project or get_fullname(frappe.session.user)
+		elif self.activity_type :
+ 			self.title = _("{0}").format(self.activity_type)
+ 		
+ 		if self.quotation_:
```
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.py]: this field is customised, we cannot use to pull to standard

File: erpnext/projects/doctype/time_log/time_log.py [hunk 4]
```diff
@@ -260,13 +266,38 @@ def get_events(start, end, filters=None):
...
 	conditions = get_event_conditions("Time Log", filters)
 
+	if (cint(get_defaults("fs_simplified_time_log"))):
+		date_cond = "date_worked between %(start)s and %(end)s"
+	else:
+		date_cond = "( from_time between %(start)s and %(end)s or to_time between %(start)s and %(end)s )"
+	
 	data = frappe.db.sql("""select name, from_time, to_time,
-		activity_type, task, project, production_order, workstation from `tabTime Log`
-		where docstatus < 2 and ( from_time between %(start)s and %(end)s or to_time between %(start)s and %(end)s )
```
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.py]: is this code necessary if "Simplified Time Log" is unchecked?

File: erpnext/projects/report/daily_time_log_summary/daily_time_log_summary.py [hunk 5]
```diff
@@ -4,49 +4,92 @@
...
-from frappe.utils import flt
+from frappe.utils import flt,cint
 
 def execute(filters=None):
+	simplified_time_log = cint(frappe.db.get_value("Project Settings", None, "simplified_time_log"))
 	if not filters:
 		filters = {}
 	elif filters.get("from_date") or filters.get("to_date"):
 		filters["from_time"] = "00:00:00"
-		filters["to_time"] = "24:00:00"
```
Comment (IAGAdmin) [inline erpnext/projects/report/daily_time_log_summary/daily_time_log_summary.py]: is this code necessary if "Simplified Time Log" is unchecked?

**Comment by IAGAdmin** [comment]
change option to "Date only" and add "To log only the Date. From and To time will be excluded" as description below

**Comment by IAGAdmin** [comment]
How does the manufacturing module interact with time log?

File: erpnext/projects/doctype/time_log/time_log.js (hunk 1 above)
Comment (ShashaQin) [inline erpnext/projects/doctype/time_log/time_log.js]: this code was from Robert, I think he didn't consider  "Simplified Time Log" unchecked situation at that time. I have to check further.

File: erpnext/projects/doctype/time_log/time_log.json (hunk 2 above)
Comment (ShashaQin) [inline erpnext/projects/doctype/time_log/time_log.json]: yes, cause standard changed some fields order, I made changes to make Time Log form look similar to before.

File: erpnext/projects/doctype/time_log/time_log.py (hunk 3 above)
Comment (ShashaQin) [inline erpnext/projects/doctype/time_log/time_log.py]: we have some customised fields, when we pull to standard, we have to do another version to add these customised fields to json file.

File: erpnext/projects/doctype/time_log/time_log.py (hunk 4 above)
Comment (ShashaQin) [inline erpnext/projects/doctype/time_log/time_log.py]: yes, necessary, this code is checking "Simplified Time Log" value and doing different condition.

File: erpnext/projects/report/daily_time_log_summary/daily_time_log_summary.py (hunk 5 above)
Comment (ShashaQin) [inline erpnext/projects/report/daily_time_log_summary/daily_time_log_summary.py]: yes, necessary, this code is checking "Simplified Time Log" value and assigning different value to filters["to_time"]

File: erpnext/projects/doctype/time_log/time_log.js (hunk 1 above)
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.js]: Ok please change it

File: erpnext/projects/doctype/time_log/time_log.json (hunk 2 above)
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.json]: Good thanks

File: erpnext/projects/doctype/time_log/time_log.py (hunk 3 above)
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.py]: Leave it for the moment as i've got an idea.

File: erpnext/projects/doctype/time_log/time_log.py (hunk 4 above)
Comment (IAGAdmin) [inline erpnext/projects/doctype/time_log/time_log.py]: ok

File: erpnext/projects/report/daily_time_log_summary/daily_time_log_summary.py (hunk 5 above)
Comment (IAGAdmin) [inline erpnext/projects/report/daily_time_log_summary/daily_time_log_summary.py]: ok

**Comment by ShashaQin** [comment]
Draft time log will be created automatically when creating production order. For detail information , please have a look the following link:
https://frappe.github.io/erpnext/user/manual/en/manufacturing/production-order.html

File: erpnext/projects/doctype/time_log/time_log.js (hunk 1 above)
Comment (ShashaQin) [inline erpnext/projects/doctype/time_log/time_log.js]: sorry, checked further, he  did check "Simplified Time Log" in update_time function that is his new function. So this code is necessary if "Simplified Time Log" is unchecked.

**Comment by ShashaQin** [comment]
update project_setting.json for change option to "Date only" and add "To log only the Date. From and To time will be excluded" as description below

**Pull request merged**